cmd/clusterctl/client/cluster/mover.go

```diff
@@ -592,6 +772,132 @@ func (o *objectMover) createTargetObject(nodeToCreate *node, toProxy Proxy) erro
 	return nil
 }
 
+func (o *objectMover) backupTargetObject(nodeToCreate *node, directory string) error {
+	log := logf.Log
+	log.V(1).Info("Saving", nodeToCreate.identity.Kind, nodeToCreate.identity.Name, "Namespace", nodeToCreate.identity.Namespace)
+
+	cFrom, err := o.fromProxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	// Get the source object
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(nodeToCreate.identity.APIVersion)
+	obj.SetKind(nodeToCreate.identity.Kind)
+	objKey := client.ObjectKey{
+		Namespace: nodeToCreate.identity.Namespace,
+		Name:      nodeToCreate.identity.Name,
+	}
+
+	if err := cFrom.Get(ctx, objKey, obj); err != nil {
+		return errors.Wrapf(err, "error reading %q %s/%s",
+			obj.GroupVersionKind(), obj.GetNamespace(), obj.GetName())
+	}
+
+	// Get JSON for object and write it into the configured directory
+	byObj, err := obj.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	filenameObj := nodeToCreate.getFilename()
+	objectFile := filepath.Join(directory, filenameObj)
+
+	// If file exists, then remove it to be written again
+	if _, err = os.Stat(objectFile); err == nil {
+		if err = os.Remove(objectFile); err != nil {
+			return err
+		}
+	}
+
+	err = ioutil.WriteFile(objectFile, byObj, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *objectMover) restoreTargetObject(nodeToCreate *node, toProxy Proxy) error {
+	log := logf.Log
+	log.V(1).Info("Restoring", nodeToCreate.identity.Kind, nodeToCreate.identity.Name, "Namespace", nodeToCreate.identity.Namespace)
+
+	// Creates the targetObj into the target management cluster.
+	cTo, err := toProxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	// Rebuild the source object
+	obj := nodeToCreate.restoreObject
+	obj.SetAPIVersion(nodeToCreate.identity.APIVersion)
+	obj.SetKind(nodeToCreate.identity.Kind)
+	objKey := client.ObjectKey{
+		Namespace: nodeToCreate.identity.Namespace,
+		Name:      nodeToCreate.identity.Name,
+	}
+
+	// New objects cannot have a specified resource version. Clear it out.
+	obj.SetResourceVersion("")
+
+	// Removes current OwnerReferences
+	obj.SetOwnerReferences(nil)
```

Comment:
What about moving this chunk of code after Get (to ~L860)?

Reply:
Sure thing - moved this below the `Get` so the two blocks are alittle more split up